### PR TITLE
feat: add batch denom link

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#964

- Add batch denom link to `send` and `retire` actions
- refactor `MyEcocredits`: split callbacks into hooks

MyEcocredits refactor

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to https://deploy-preview-993--regen-registry.netlify.app/ecocredits/dashboard
2. Send or retire a credit
3. Check denom link in confirmation modal

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/pages/MyEcocredits/hooks/useCreditRetireSubmit.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        fee: undefined,
+        memo: values?.note,
+      };
+
+      await signAndBroadcast(tx, () => setCreditRetireOpen(-1));
+      if (batchDenom && amount) {
+        setCardItems([
+          {
+            label: 'batch denom',
+            value: { name: batchDenom, url: `/credit-batches/${batchDenom}` },
```
Comment (flagrede) [inline web-registry/src/pages/MyEcocredits/hooks/useCreditRetireSubmit.tsx:63]: Actual change of denom link for retire action.

File: web-registry/src/pages/MyEcocredits/hooks/useCreditSendSubmit.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        fee: undefined,
+        memo: values?.note,
+      };
+
+      await signAndBroadcast(tx, () => setCreditSendOpen(-1));
+      if (batchDenom && recipient) {
+        setCardItems([
+          {
+            label: 'batch denom',
+            value: { name: batchDenom, url: `/credit-batches/${batchDenom}` },
```
Comment (flagrede) [inline web-registry/src/pages/MyEcocredits/hooks/useCreditSendSubmit.tsx:65]: Actual change of denom link for send action.

File: web-registry/src/hooks/useMsgClient.ts [hunk 3]
```diff
@@ -11,11 +11,13 @@ interface TxData {
   memo?: string;
 }
 
+export type SignAndBroadcastType = (
```
Comment (flagrede) [inline web-registry/src/hooks/useMsgClient.ts:14]: Make signAndBroadcast type reusable.

File: web-registry/src/types/react/use-state.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import { Dispatch, SetStateAction } from 'react';
+
+export type useStateSetter<T> = Dispatch<SetStateAction<T>>;
```
Comment (flagrede) [inline web-registry/src/types/react/use-state.ts:3]: Small generic util for React useState type.

File: web-registry/src/pages/MyEcocredits/MyEcocredits.tsx [hunk 5]
```diff
@@ -22,54 +12,58 @@ import {
...
+import useQueryBaskets from '../../hooks/useQueryBaskets';
+import { useLedger } from '../../ledger';
+import { getHashUrl } from '../../lib/block-explorer';
+import useBasketPutSubmit from './hooks/useBasketPutSubmit';
+import useBasketTakeSubmit from './hooks/useBasketTakeSubmit';
+import useCreditRetireSubmit from './hooks/useCreditRetireSubmit';
+import useCreditSendSubmit from './hooks/useCreditSendSubmit';
+import { default as useStyles } from './hooks/useMyEcocreditsStyles';
+import useOpenTakeModal from './hooks/useOpenTakeModal';
+import useUpdateCrediBaskets from './hooks/useUpdateCrediBaskets';
```
Comment (blushi) [inline web-registry/src/pages/MyEcocredits/MyEcocredits.tsx:50]: small typo
```suggestion
import useUpdateCreditsBaskets from './hooks/useUpdateCreditsBaskets';
```

File: web-registry/src/pages/MyEcocredits/MyEcocredits.tsx [hunk 6]
```diff
@@ -102,206 +95,78 @@ export const MyEcocredits = (): JSX.Element => {
     error,
     setError,
   } = useMsgClient(handleTxQueued, handleTxDelivered, handleError);
+
+  const { api } = useLedger();
```
Comment (blushi) [inline web-registry/src/pages/MyEcocredits/MyEcocredits.tsx:99]: I believe we could actually get rid of this, it's only used for checking `api?.msgClient` in the hooks, which is already handled in `useMsgClient`

**Review by blushi — APPROVED** [review]
tACK, thanks for the refactor! pre-approving, pending small nits

**Comment by flagrede** [comment]
@erikalogie I've added testing instructions in the description for the QA.

File: web-registry/src/pages/MyEcocredits/MyEcocredits.tsx (hunk 5 above)
Comment (flagrede) [reply web-registry/src/pages/MyEcocredits/MyEcocredits.tsx:50]: Nice catch !

File: web-registry/src/pages/MyEcocredits/MyEcocredits.tsx (hunk 6 above)
Comment (flagrede) [reply web-registry/src/pages/MyEcocredits/MyEcocredits.tsx:99]: You're right I will remove it.

**Comment by erikalogie** [comment]
Looks good to me! Side note I just noticed that the recipient does not link anywhere which does not match the designs so I opened a ticket for it: #982.

**Comment by victordelval** [comment]
@flagrede @blushi LGTM... but I have one important question, and that is that we are introducing another pattern to organize the code by making pages modules/folders. I think we have to come to a consensus on this. 

Until recently, file-like container components were growing indistinctly both in `/components/organisms` and in `/pages`. We've been talking for the last few weeks about starting to make these modules or directories so that we can refactor these large components into their own directory. 

I was proposing to start a new structure from the root that was for example `/features`. Marie I think she told me that she preferred `/components/features` and now I see that with this PR a third way is being introduced which is to take it directly to `/pages`...

The solutions are similar, although each one has its nuances. I think we should just talk it over together and agree or something, right?

**Review by victordelval — APPROVED** [review]

**Comment by flagrede** [comment]
> @flagrede @blushi LGTM... but I have one important question, and that is that we are introducing another pattern to organize the code by making pages modules/folders. I think we have to come to a consensus on this.
> 
> Until recently, file-like container components were growing indistinctly both in `/components/organisms` and in `/pages`. We've been talking for the last few weeks about starting to make these modules or directories so that we can refactor these large components into their own directory.
> 
> I was proposing to start a new structure from the root that was for example `/features`. Marie I think she told me that she preferred `/components/features` and now I see that with this PR a third way is being introduced which is to take it directly to `/pages`...
> 
> The solutions are similar, although each one has its nuances. I think we should just talk it over together and agree or something, right?

@victordelval I agree with you that we should agree on a team convention on this. That being said I think the file would have needed the split wherever the files are located. Happy to move/update the files once we agree on a convention.

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**